tests/Application/src/Tests/Controller/SubscriptionJsonApiTest.php

```diff
@@ -47,41 +57,140 @@ public function it_allows_indexing_subscriptions(): void
+                        "email": "doc.brown@bttf.com"
+                    },
+                    {
+                        "state": "accepted",
+                        "email": "biff.tannen@bttf.com"
+                    },
+                    {
+                        "state": "new",
+                        "email": "lorraine.baines@bttf.com"
+                    },
+                    {
+                        "state": "new",
+                        "email": "george.mcfly@bttf.com"
+                    },
+                    {
+                        "state": "new",
+                        "email": "jennifer.parker@bttf.com"
+                    }
+                ],
+                "pagination": {
+                    "current_page": 1,
+                    "has_previous_page": false,
+                    "has_next_page": false,
+                    "per_page": 10,
+                    "total_items": 6,
+                    "total_pages": 1
+                }
+            }
+            JSON
+        );
     }
 
     #[Test]
     public function it_allows_creating_a_subscription(): void
     {
         $data =
-            <<<EOT
-        {
-            "email": "marty.mcfly@bttf.com"
-        }
-EOT;
+            <<<'JSON'
+            {
+                "email": "marty.mcfly@bttf.com"
+            }
+            JSON
+        ;
 
         $this->client->request('POST', '/ajax/subscriptions', [], [], ['CONTENT_TYPE' => 'application/json'], $data);
-        $response = $this->client->getResponse();
-        $this->assertResponse($response, 'subscriptions/create_response', Response::HTTP_CREATED);
+
+        $this->assertResponseIsSuccessful();
+        $this->assertResponseStatusCodeSame(Response::HTTP_CREATED);
+        $this->assertResponseHeaderSame('content-type', 'application/json; charset=utf-8');
+
+        $this->assertResponseMatchesPattern(
+            <<<'JSON'
+            {
+                "state": "new",
+                "email": "marty.mcfly@bttf.com"
+            }
+            JSON
+        );
     }
 
     #[Test]
     public function it_does_not_allow_to_create_a_subscription_if_there_is_a_validation_error(): void
     {
         $data =
-            <<<EOT
-        {
-            "email": ""
-        }
-EOT;
+            <<<'JSON'
+            {
+                "email": ""
+            }
+            JSON
+        ;
 
         $this->client->request('POST', '/ajax/subscriptions', [], [], ['CONTENT_TYPE' => 'application/json'], $data);
 
-        $file = Kernel::VERSION_ID >= 60400 ? 'subscriptions/create_validation' : 'subscriptions/create_validation_legacy';
+        $this->assertResponseStatusCodeSame(Response::HTTP_UNPROCESSABLE_ENTITY);
+        $this->assertResponseHeaderSame('content-type', 'application/json; charset=utf-8');
 
-        $this->assertResponse($this->client->getResponse(), $file, Response::HTTP_UNPROCESSABLE_ENTITY);
+        if (Kernel::VERSION_ID >= 60400) {
```

Comment:
I think it would made sense to switch places for new and legacy responses. Default response should be new one and legacy should be hidden behind `if () {}`

Reply:
yeah, I agree

Reply:
Done.